packages/api/src/common/transformers/hexToDecimalNumber.transformer.spec.ts

```diff
@@ -9,7 +9,7 @@ describe("hexToDecimalNumberTransformer", () => {
 
     it("returns hex representation of the decimal number string", () => {
       const result = hexToDecimalNumberTransformer.to("800");
```

Comment:
That's a weird transformer. Should ethers provide something like that already?
And also it's weird that the value changed. Was it incorrect before?

Reply:
Yeah, we need this one to transform decimal string to a hex string. Value has changed from `0x0320` to `0x320` which is the same value in the hex representation:
```
> BigInt("0x0320") === BigInt("0x320")
true
```
it's the same as with numbers in the decimal number system `800 === 0800`, so leading zeros can be ignored. Why the hex representation changed: old SDK uses `ethers v5` with `BigNumber` lib under the hood. In `ethers v6`, which is used by the new SDK, big numbers are now represented by the build in `bigint` type. `BigNumber` `toHexString` returned the value with one leading zero, while `bigint` `toString(16)` does not add any leading zeros.